server/core/hot_region_storage.go

```diff
@@ -171,13 +167,17 @@ func (h *HotRegionStorage) backgroundDelete() {
 		h.hotRegionLoopWg.Done()
 	}()
 	for {
+		reserverDays := int(h.hotRegionStorageHandler.GetHotRegionsReservedDays())
 		select {
 		case <-ticker.C:
 			if isFirst {
 				ticker.Reset(24 * time.Hour)
 				isFirst = false
 			}
-			h.delete()
+			if reserverDays == 0 {
+				continue
+			}
+			h.delete(reserverDays)
```

Comment:
Unhandled error. How about adding a warn log?

Reply:
How about call `delete` too when `reserverDays` is 0, it seems to be ok?

Reply:
How about call `delete` too when `reserverDays` is 0, it seems to be ok?

Reply:
We have added a warning log here. please see if it fits